docs/book/user-guide/production-guide/ci-cd.md

```diff
@@ -6,28 +6,135 @@ description: >-
+
+This step is optional, all you'll need for certain is a stack that runs remotely (remote orchestration and artifact
+storage). The rest is up to you. You might for example want to parametrize your pipeline to use different data sources
+for the respective environments. You can also use different [configuration files](../../how-to/configuring-zenml/configuring-zenml.md)
+for the different environments to configure the [Model](../../how-to/use-the-model-control-plane/README.md), the 
+[DockerSettings](../../how-to/customize-docker-builds/docker-settings-on-a-pipeline.md), the [ResourceSettings like
+accelerators](../../how-to/training-with-gpus/training-with-gpus.md) differently for the different environments.
+
+## Run Pipeline for PRs
+
+One way to ensure only fully working code makes it into production, you should use a staging environment to test all
+the changes made to your code base and verify they work as intended. To do so automatically you should set up a
+github action workflow that runs your pipeline for you when you make changes to it.
+[Here](https://github.com/zenml-io/zenml-gitflow/blob/main/.github/workflows/pipeline_run.yaml) is an example that you
+can use.
+
+To only run the Github Action on a PR, you can configure the yaml like this
+
+```yaml
+on:
+  pull_request:
+    branches: [ staging, main ]
+```
+
+When the workflow starts we want to set some important values. Here is a simplified version that you can use.
+
+```yaml
+jobs:
+  run-staging-workflow:
+    runs-on: run-zenml-pipeline
+    env:
+      ZENML_HOST: ${{ secrets.ZENML_HOST }}  # Put your server url here
+      ZENML_API_KEY: ${{ secrets.ZENML_API_KEY }}  # Retrieves the api key for use  
+      ZENML_STACK: stack_name  #  Use this to decide which stack is used for staging
+      ZENML_GITHUB_SHA: ${{ github.event.pull_request.head.sha }}
+      ZENML_GITHUB_URL_PR: ${{ github.event.pull_request._links.html.href }}
+```
+
+After configuring these values so they apply to your specific situation the rest of the template should work as is for
+you. Specifically you will need to install all requirements, connect to your ZenML Server, set an active stack
+and run a pipeline within your github action.
+
+```yaml
+steps:
+  - name: Check out repository code
+    uses: actions/checkout@v3
+
+  - uses: actions/setup-python@v4
+    with:
+      python-version: '3.9'
+
+  - name: Install requirements
+    run: |
+      pip3 install -r requirements.txt
+
+  - name: Connect to ZenML server
+    run: |
+      zenml connect --url $ZENML_HOST --api-key $ZENML_API_KEY
+
+  - name: Set stack
+    run: |
+      zenml stack set ${{ env.ZENML_STACK }}
+
+
+  - name: Run pipeline
+    run: |
+      python run.py \
+        --pipeline end-to-end \
+        --dataset production \
+        --version ${{ env.ZENML_GITHUB_SHA }} \
+        --github-pr-url ${{ env.ZENML_GITHUB_URL_PR }}
+```
```

Comment:
might make sense to end with a screenshot of a running github action or even a PR comment?

Reply:
So are you gonna add it? :D